sky/resources.py

```diff
@@ -275,7 +275,49 @@ def _set_region_zone(self, region: Optional[str],
         self._region, self._zone = self._cloud.validate_region_zone(
             region, zone)
 
-    def _try_validate_instance_type(self):
+    def get_valid_region_zones_for_launchable(
+            self) -> List[Tuple[clouds.Region, List[clouds.Zone]]]:
+        """Returns a set of (region, zone) that can provision this Resources."""
+        assert self.is_launchable()
+
+        # Get the resource-offering regions and zones.
+        region_zones = list(
+            self._cloud.region_zones_provision_loop(
```

Comment:
Do you think we can change the return value of this function and `self._cloud.region_zones_provision_loop` to only return `List[Region]`? It seems a bit weird to me to return a tuple of `region_zone`, where the `region_zone[0].zones` can conflict with `region_zone[1]`.

Reply:
I feel it can be ambiguous if a `Region` object has only a subset of its `Zone`s. I can change the return type of `get_valid_region_zones_for_launchable` from `List[Tuple[clouds.Region, List[clouds.Zone]]]` to `Dict[str, List[str]]` as @concretevitamin suggested. WDYT?